mobile/lib/widgets/settings/networking_settings/endpoint_input.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+import 'package:immich_mobile/models/auth/auxilary_endpoint.model.dart';
+import 'package:immich_mobile/providers/auth.provider.dart';
+import 'package:immich_mobile/widgets/settings/networking_settings/networking_settings.dart';
+
+class EndpointInput extends StatefulHookConsumerWidget {
+  const EndpointInput({
+    super.key,
+    required this.initialValue,
+    required this.index,
+    required this.onValidated,
+    required this.onDismissed,
+  });
+
+  final AuxilaryEndpoint initialValue;
+  final int index;
+  final Function(String url, int index, AuxCheckStatus status) onValidated;
+  final Function(int index) onDismissed;
+
+  @override
+  EndpointInputState createState() => EndpointInputState();
+}
+
+class EndpointInputState extends ConsumerState<EndpointInput> {
+  late final TextEditingController controller;
+  late final FocusNode focusNode;
+  late AuxCheckStatus auxCheckStatus;
+  bool isInputValid = false;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = TextEditingController(text: widget.initialValue.url);
+    focusNode = FocusNode()..addListener(_onOutFocus);
+
+    setState(() {
+      auxCheckStatus = widget.initialValue.status;
+    });
+  }
+
+  @override
+  void dispose() {
+    focusNode.removeListener(_onOutFocus);
+    focusNode.dispose();
+    controller.dispose();
+    super.dispose();
+  }
+
+  void _onOutFocus() {
+    if (!focusNode.hasFocus && isInputValid) {
+      validateAuxilaryServerUrl();
+    }
+  }
+
+  Future<void> validateAuxilaryServerUrl() async {
+    final url = controller.text;
+    setState(() => auxCheckStatus = AuxCheckStatus.loading);
+
+    final isValid =
+        await ref.read(authProvider.notifier).validateAuxilaryServerUrl(url);
+
+    setState(() {
```

Comment:
It is a good practise to check if the widget is still mounted between async gaps

Reply:
You mean to add `if (mounted) {}`?

Reply:
Yes, if the widget is unmounted when the asynchronous call was in progress, the next setState will throw an error

Reply:
Would you suggest to check this in all `setState` call?

Reply:
AFAIK it's needed in async functions and callbacks, but isn't needed outside of those cases.